tests/unit-tests/mailpoet/test-class-sensei-mailpoet.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * File with class for testing Sensei MailPoet.
+ *
+ * @package sensei-tests
+ */
+
+// phpcs:disable WordPress.NamingConventions.ValidFunctionName.MethodNameInvalid -- Mocking an external library.
+// phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound -- Using PHPUnit conventions.
+
+/**
+ * Class for testing Sensei_MailPoet class.
+ *
+ * @group Sensei MailPoet
+ */
+class Sensei_MailPoet_Test extends WP_UnitTestCase {
+	use Sensei_Course_Enrolment_Manual_Test_Helpers;
+	/**
+	 * Factory object.
+	 *
+	 * @var Sensei_Factory
+	 */
+	protected $factory;
+
+	/**
+	 * Set up the test.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->factory = new Sensei_Factory();
+		$mailpoet_api  = Sensei_MailPoetAPIMockFactory::MP();
```

Comment:
Great job finding this class! Seems like the best class that would solve all our problems regarding the previous discussion we had, on mocking keys, etc. :)